src/main/java/org/sonarlint/intellij/analysis/LocalFileExclusions.java

```diff
@@ -221,12 +221,21 @@ private ExcludeResult checkProjectStructureExclusion(VirtualFile file) {
   @NotNull
   private static ExcludeResult excludeUnsupportedFileOrFileType(VirtualFile file) {
     var fileType = file.getFileType();
-    if (!file.isInLocalFileSystem() || fileType.isBinary() || !file.isValid() || ".idea".equals(file.getParent().getName())) {
+    if (!file.isInLocalFileSystem() || fileType.isBinary() || !file.isValid() ||
+      ".idea".equals(file.getParent().getName()) || isRazorFile(file)) {
       return ExcludeResult.excluded("file's type or location are not supported");
     }
     return ExcludeResult.notExcluded();
   }
 
+  public static boolean isRazorFile(VirtualFile file) {
+    if (file.getExtension() == null) {
+      return false;
+    }
+
+    return file.getExtension().equals("razor");
```

Comment:
can be simplified to:
`return file.getExtension() != null && file.getExtension().equals("razor");`